azurelinuxagent/common/cgroup.py

```diff
@@ -213,26 +214,29 @@ def get_cpu_usage(self):
 
         return round(100.0 * self._osutil.get_processor_cores() * float(cgroup_delta) / float(system_delta), 3)
 
+    def get_throttled_time(self):
+        """
+        Computes the throttled time (in nanoseconds) since the last call to this function.
+        NOTE: initialize_cpu_usage() must be invoked before calling this function
+        """
+        if not self._cpu_usage_initialized():
+            raise CGroupsException("initialize_cpu_usage() must be invoked before the first call to get_throttled_time()")
+
+        self._previous_throttled_time = self._current_throttled_time
+        self._current_throttled_time = self._get_throttled_time()
+
+        return  self._current_throttled_time - self._previous_throttled_time
```

Comment:
Is this always guaranteed to be incrementing? Like at current would always be > previous? 
Incase of filenotfound, current would be 0. Is there a possibility that this file can be deleted somehow externally?

Reply:
yes, it is incremental from the creation of the cgroup

Reply:
oh, just noted your question about deletion

cgroups it's a "pseudo" file system. can't delete the file (there might be a way to do it, but straight rm doesn't do it):

root@nam-u18:/var/lib/waagent# rm /sys/fs/cgroup/cpuacct/azure.slice/cpu.stat 
rm: cannot remove '/sys/fs/cgroup/cpuacct/azure.slice/cpu.stat': Operation not permitted

Reply:
woah! That is interesting!